libraries/Microsoft.Bot.Builder.Alexa/Integration/AspNet.Core/AlexaRequestHandler.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Bot.Builder.Alexa.Integration.AspNet.Core
+{
+    public class AlexaRequestHandler
+    {
+        public static readonly JsonSerializer AlexaBotMessageSerializer = JsonSerializer.Create(new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore,
+        });
+
+        private readonly AlexaAdapter _alexaAdapter;
+
+        public AlexaRequestHandler(AlexaAdapter alexaAdapter)
+        {
+        }
+       
+        protected async Task ProcessMessageRequestAsync(HttpRequest request, AlexaAdapter alexaAdapter, Func<ITurnContext, Task> botCallbackHandler)
+        {
+            AlexaRequestBody skillRequest;
+            byte[] requestByteArray;
+
+            //if (_validateIncomingAlexaRequests)
+            //{
+            //var requestValidationHelper = new AlexaRequestValidationHelper();
+            //await requestValidationHelper.ValidateRequestSecurity(request, requestByteArray, skillRequest);
+            //}
+
+            using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8)))
+            {
+                skillRequest = AlexaBotMessageSerializer.Deserialize<AlexaRequestBody>(bodyReader);
+            }
+
+            if (skillRequest.Version != "1.0")
+                throw new Exception($"Unexpected version of '{skillRequest.Version}' received.");
+
+            await alexaAdapter.ProcessActivity(
```

Comment:
>ProcessActivity [](start = 31, length = 15)

there is no reason the signature of this couldn't be that it takes the SkillRequest as input